rotkehlchen/accounting/cost_basis/base.py

```diff
@@ -364,12 +366,16 @@ def add_acquisition(self, acquisition: AssetAcquisitionEvent) -> None:
             self._acquisitions_heap,
             AssetAcquisitionHeapElement(self._count, acquisition),
         )
-        self.current_average_cost_basis += (acquisition.rate * acquisition.amount)
+        self.current_average_asset_value = (self.current_average_asset_value * self.remaining_amount + acquisition.amount * acquisition.rate) / (self.remaining_amount + acquisition.amount)  # noqa: E501
         self.remaining_amount += acquisition.amount
         self._count += 1
 
     def consume_result(self, used_amount: FVal) -> None:
-        """Same as its parent function but also deducts `used_amount` from `remaining_amount`."""
+        """
+        Same as its parent function but also deducts `used_amount` from `remaining_amount`.
+        This function is supposed to always be called under `processing_iterator`, which guarantees
+        that the remaining amount is greater than zero.
```

Comment:
nitpick: Since the division by remaining amount is gone from here this comment is no longer needed

```suggestion
```